docs/installation/server-config/security.rst

```diff
@@ -132,28 +133,130 @@ Configuring Authorization of KSQL with Kafka ACLs
+
+How you define ACLs to allow KSQL access to the input and output topics will depend on your use case and
+whether the KSQL cluster is configured for
+:ref:`interactive <config-security-ksql-acl-interactive_post_ak_2_0>` or :ref:`non-interactive (headless) <config-security-ksql-acl-headless_post_ak_2_0>`
+
+.. _config-security-ksql-acl-headless_post_ak_2_0:
+
+Non-Interactive (headless) KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+:ref:`Non-interactive KSQL clusters <restrict-ksql-interactive>` run a known set of SQL statements, meaning the set
+of input and output topics is well defined. Add the ACLs required to allow KSQL access to these topics.
+
+.. config-security-ksql-acl-interactive_post_ak_2_0
+
+Interactive KSQL clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:ref:`Interactive KSQL clusters <restrict-ksql-interactive>` accept SQL statements from users and hence may require access
+to a wide variety of input and output topics. Add ACLs to appropriate literal and prefixed resource patterns to allow KSQL
+access to the input and output topics, as required.
+
+As the set of output topics may not be known up front, it is common to to add an ACL to allow ``ALL`` operations
+on ``TOPIC``s ``PREFIXED`` with some chosen prefix, which will be used for all output topics in the cluster.
+Thereby allowing users to create output topics as required.
+The ``ALL`` operations allows topics to be created on-demand and to be used as input topics for subsequent statements.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Kafka Cluser versions below v2.0
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Versions of Kafka below v2.0 do not benefit from the enhancements found in later version of Kafka that simplify the ACLs required to run
+KSQL against a Kafka cluster secured with ACLs.
+The list of ACLs that must be defined depends on whether the KSQL cluster is configured for
+:ref:`interactive <config-security-ksql-acl-interactive_pre_ak_2_0>` or :ref:`non-interactive (headless) <config-security-ksql-acl-headless_pre_ak_2_0>`.
+
+ACL definition
+^^^^^^^^^^^^^^
+
+Kafka acls are defined in the general format of "Principal P is [Allowed/Denied] Operation O From Host H on Resource R".
+
+Principal
+    An authenticated user or group. For example, ``"user: Fred"`` or ``"group: fraud"``.
 
 Permission
     Defines if the ACL allows (``ALLOW``) or denies (``DENY``) access to the resource.
 
-Principal
-    An authenticated user or group. For example, ``"user: Fred"`` or ``"group: fraud"``. 
+Operation
+    The operation that is performed on the resource, for example ``READ``.
+
+Resource
+    A resource is comprised of a resource type and resource name:
+
+    - ``RESOURCE_TYPE``, for example ``TOPIC`` or consumer ``GROUP``.
+    - Resource name, where the name is either specific, e.g. ``users``, or the wildcard ``*``, meaning all resources of this type.
+
+    The ``CLUSTER`` resource type does not require a resource name because it refers to the entire Kafka cluster.
```

Comment:
Similar to @apurvam 's comment above, can we also show example admin commands to set the (here: rather broad) ACLs?